src/core/thread/router_table.cpp

```diff
@@ -306,7 +306,7 @@ otError RouterTable::Release(uint8_t aRouterId)
     return error;
 }
 
-void RouterTable::RemoveNeighbor(Router &aRouter)
```

Comment:
Is this change related to this PR?

Reply:
It might be not so related to the intention of this PR, however `RemoveNeighbor()` name does not exactly reflect what it does when updating `ProcessTlv()` to `ProcessRouterIdSet()`.
It is just a small piece of code, prefer to keep it as is.

Reply:
Agree that `RouterTable::RemoveNeighbor` does not reflect what it actually do. But, neither do I think `RemoveRouter` reflects exactly. 
`RemoveRouter` is basically same as `RemoveNeighbor`, because every `Neighbor` in `RouterTable` is `Router`. However, what this function actually removes is the **ROUTES** that use the router as the next hop. 

So, maybe this function should be renamed to `RemoveRoutes` or `RemoveRoutesFrom`? Thoughts? @jwhui @bukepo

Reply:
FWIW, `RemoveNeighbor()` was closer than `RemoveRouter()`. This method removes direct link to the neighboring device (effectively removing the neighbor). Of course, as a result, this also removes any routes that utilize the direct neighbor link.

Reply:
The removed entry is what was in the router table but now not when learning the latest `RouterIdSet` either from `RouteTlv` or `ThreadRouterMaskTlv`, it may be the device's direct neighbor, but may also be some other routers which were routable.

As @jwhui , when removing one router entry, the other entries which was routed via this entry should be updated along with.

Thoughts?

Reply:
This method is also in the call chain initiated by `MeshForwarder()` to invalidate a link to a neighboring device.
https://github.com/openthread/openthread/blob/d85c0007a13b05ceac5f2b070417d9f191bc5cdc/src/core/thread/mesh_forwarder.cpp#L813

If we want to rename, we could adopt the same name `RemoveRouterLink()`.

Reply:
Thought the neighboring router also belongs router, not conflicting with `RemoveRouter()` name.
Not sure but the `Link` in `RemoveRouterLink()` gives me the impression is it is removing direct neighbor, is it wrong?

Reply:
This method only sets the link quality to zero and invalidates any routes that utilize the link. It does not actually remove any other neighbor or router state. `RemoveRouter()` might give the impression that the frame counters and other state are being removed. Thoughts?

Reply:
ok. that makes sense. will update.